clouddriver-google/src/main/groovy/com/netflix/spinnaker/clouddriver/google/provider/agent/GoogleInternalLoadBalancerCachingAgent.groovy

```diff
@@ -0,0 +1,365 @@
+    List<GoogleInternalLoadBalancer> loadBalancers = []
+
+    BatchRequest forwardingRulesRequest = buildBatchRequest()
+    BatchRequest backendServiceRequest = buildBatchRequest()
+    BatchRequest groupHealthRequest = buildBatchRequest()
+    BatchRequest healthCheckRequest = buildBatchRequest()
+
+    ForwardingRulesCallback callback = new ForwardingRulesCallback(
+      loadBalancers: loadBalancers,
+      backendServiceRequest: backendServiceRequest,
+      healthCheckRequest: healthCheckRequest,
+      groupHealthRequest: groupHealthRequest,
+    )
+    compute.forwardingRules().list(project, region).queue(forwardingRulesRequest, callback)
+
+    executeIfRequestsAreQueued(forwardingRulesRequest)
+    executeIfRequestsAreQueued(backendServiceRequest)
+    executeIfRequestsAreQueued(healthCheckRequest)
+    executeIfRequestsAreQueued(groupHealthRequest)
+
+    return loadBalancers
+  }
+
+  CacheResult buildCacheResult(ProviderCache _, List<GoogleInternalLoadBalancer> googleLoadBalancers) {
+    log.info "Describing items in ${agentType}"
+
+    def cacheResultBuilder = new CacheResultBuilder()
+
+    googleLoadBalancers.each { GoogleInternalLoadBalancer loadBalancer ->
+      def loadBalancerKey = Keys.getLoadBalancerKey(
+        loadBalancer.region,
+        loadBalancer.account,
+        loadBalancer.name
+      )
+      def instanceKeys = loadBalancer.healths.collect { GoogleLoadBalancerHealth health ->
+        Keys.getInstanceKey(accountName, loadBalancer.region, health.instanceName)
+      }
+
+      cacheResultBuilder.namespace(LOAD_BALANCERS.ns).keep(loadBalancerKey).with {
+        attributes = objectMapper.convertValue(loadBalancer, ATTRIBUTES)
+      }
+      instanceKeys.each { String instanceKey ->
+        cacheResultBuilder.namespace(INSTANCES.ns).keep(instanceKey).with {
+          relationships[LOAD_BALANCERS.ns].add(loadBalancerKey)
+        }
+      }
+    }
+
+    log.info "Caching ${cacheResultBuilder.namespace(LOAD_BALANCERS.ns).keepSize()} load balancers in ${agentType}"
+    log.info "Caching ${cacheResultBuilder.namespace(INSTANCES.ns).keepSize()} instance relationships in ${agentType}"
+
+    cacheResultBuilder.build()
+  }
+
+  @Override
+  boolean handles(OnDemandAgent.OnDemandType type, String cloudProvider) {
+    type == OnDemandAgent.OnDemandType.LoadBalancer && cloudProvider == GoogleCloudProvider.GCE
+  }
+
+  @Override
+  Collection<Map> pendingOnDemandRequests(ProviderCache providerCache) {
+    return []
+  }
+
+  /**
+   * This is a "simpleton" way of handling on-demand cache requests. Load Balancer mutation (and thus the need for
+   * cache refreshing) is not as common or complex as server group cache refreshes.
+   *
+   * This implementation has the potential for race condition between handle() and loadData(), which may
+   * cause "flapping" in the UI. lwander@ has plans to make an abstract solution for this race condition, so this impl
```

Comment:
I've seen this note in earlier PR's as well. Please ensure we have a bug to track that fix.